libs/pmp-web/repository/data-access/src/lib/+state/repository.facade.ts

```diff
@@ -26,12 +29,14 @@ export class RepositoryFacade {
     this.store.dispatch(new fromRepositoryActions.GetRepositoryCollection());
   }
 
-  addRepository(data: AddRepositoryPayload): Observable<void> {
-    return this.actionStatusResolverService.resolve(
-      new fromRepositoryActions.AddRepository(data),
-      fromRepositoryActions.Types.AddRepositorySuccess,
-      fromRepositoryActions.Types.AddRepositoryFail
-    );
+  addRepository(data: AddRepositoryPayload): Observable<Repository> {
+    return this.actionStatusResolverService
+      .resolve<AddRepositorySuccessPayload>(
+        new fromRepositoryActions.AddRepository(data),
+        fromRepositoryActions.Types.AddRepositorySuccess,
+        fromRepositoryActions.Types.AddRepositoryFail
+      )
+      .pipe(map(payload => payload.repository));
```

Comment:
from the facade please return Observable<AddRepositorySuccessPayload>. Don't think about the usage yet.